beets/dbcore/db.py

```diff
@@ -267,14 +280,14 @@ class Model:
     """
 
     @classmethod
-    def _getters(cls):
+    def _getters(cls: Type['Model']):
         """Return a mapping from field names to getter functions.
         """
         # We could cache this if it becomes a performance problem to
         # gather the getter mapping every time.
         raise NotImplementedError()
 
-    def _template_funcs(self):
+    def _template_funcs(self) -> NoReturn:
```

Comment:
Is it really correct to type this as `NoReturn`, when it is expected to be overridden with a function `-> Mapping[str, Callable[[str], str]]`   ? (not sure that I got the callable right, should check actual users of `_template_funcs`)

Reply:
I honestly don't know the conventions here. It might be a matter of preference and style whether to mark functions that should be overloaded. Perhaps it will help for IDEs/linters? It's a minor thing, but the typing does exist and this is its explicit purpose in the Python docs.

Reply:
Yes, I don't know either what the "recommended" approach is. However, in the present case of an abstract method it seems more natural to me to declare the interface that should be implemented, rather than that of the stub method in the super class. I couldn't really find many examples for this; the mypy docs do something similar at https://mypy.readthedocs.io/en/stable/error_code_list.html?highlight=notimplementederror#check-that-call-to-an-abstract-method-via-super-is-valid-safe-super